test/test_einsum.py

```diff
@@ -56,6 +53,9 @@ def naive_plated_einsum(eqn, *terms, **kwargs):
     "a,a->a",
     "a,a,a,ab->ab",
     "a,ab,bc,cd->",
+    make_hmm_einsum(10),
+    # make_hmm_einsum(15),  # slows down tests
+    # make_hmm_einsum(20),  # slows down tests
```

Comment:
I don't think the optimizer is working correctly yet, it's failing to produce optimized expressions for these examples.